File: tps2/tp2/somadigitos.c
/**
 *
 * Pontificia Universidade Catolica de Minas Gerais
 *
 * Curso de Ciencia da Computacao
 * Algoritimo e Estruturas de Dados II
 *
 * TP02Q04 - Soma de Dígitos - v1.0 - 04 / 09 / 2025
 * 855796 - Rubens Dias Bicalho
 *
 */

#include <stdio.h>
#include <string.h>
#include <stdlib.h>

// Função recursiva que retorna a soma dos dígitos de num
int somaDigitos(int num)
{
  if (num == 0)
  {
    return 0;
  }
  return (num % 10) + somaDigitos(num / 10);
}

int main()
{
  char input[100];

  while (1)
  {
    // Lê a linha de entrada
    if (!fgets(input, sizeof(input), stdin))
      break;

    // Remove a quebra de linha
    int len = strlen(input);
    if (len > 0 && input[len - 1] == '\n')
      input[len - 1] = '\0';

    // Verifica se é "FIM"
    if (strcmp(input, "FIM") == 0)
      break;

    // Converte para inteiro
    int num = atoi(input);

    // Chama a função recursiva e mostra o resultado
    printf("%d\n", somaDigitos(num));
  }

  return 0;
}

File: tps_old/tp1/ex08_somaDigitos/SomaDigitos.java

/**
 *
 * Pontificia Universidade Catolica de Minas Gerais
 *
 * Curso de Ciencia da Computacao
 * Algoritimo e Estruturas de Dados II
 *
 * TP01Q08 - Soma de Dígitos - v1.0 - 08 / 03 / 2025
 * 855796 - Rubens Dias Bicalho
 *
 */

import java.util.Scanner;

public class SomaDigitos {
    public static int soma(int input) {
        if (input == 0) {
            return 0;
        }
        return (input % 10) + soma(input / 10);
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        do {
            int num = Integer.parseInt(input);
            System.out.println(soma(num));
            input = scanner.nextLine();
        }
        while (!input.equals("FIM"));
        scanner.close();
    }

}
